### PR TITLE
schedule: move all NewOperator into operator

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
We have a lot of operators right now, and these operators are created through API, schedulers and checkers. We use `NewOperator` everywhere and each of them needs `steps` as the parameters, which is complex.

### What is changed and how it works?
Since we already have some functions like `CreateMovePeerOperator`. This PR abstracts more functions to create corresponding operators and makes the code unified.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/schedule/operator.go [hunk 1]
```diff
@@ -647,6 +726,76 @@ func getIntersectionStores(a []*metapb.Peer, b []*metapb.Peer) map[uint64]struct
 	return intersection
 }
 
+// CreateScatterRegionOperator creates an operator that scatters the specified region.
+func CreateScatterRegionOperator(desc string, cluster Cluster, origin *core.RegionInfo, replacedPeers, targetPeers []*metapb.Peer) *Operator {
```
Comment (nolouch) [inline server/schedule/operator.go]: This operator is not very general, maybe it's more readable in the old place?

File: server/schedule/operator.go [hunk 2]
```diff
@@ -462,7 +485,42 @@ func CreateAddPeerSteps(newStore uint64, peerID uint64, cluster Cluster) []Opera
...
+
+// CreateMoveRegionOperator creates an operator that moves a region to specified stores.
+func CreateMoveRegionOperator(desc string, cluster Cluster, region *core.RegionInfo, kind OperatorKind, storeIDs map[uint64]struct{}) (*Operator, error) {
+	var steps []OperatorStep
+	// Add missing peers.
+	for id := range storeIDs {
+		peer, err := cluster.AllocPeer(id)
+		if err != nil {
+			return nil, err
+		}
```
Comment (nolouch) [inline server/schedule/operator.go]: what if the peer already in the `storeIDs` or store is down?

File: server/schedule/operator.go [hunk 3]
```diff
@@ -462,7 +485,42 @@ func CreateAddPeerSteps(newStore uint64, peerID uint64, cluster Cluster) []Opera
...
+			steps = append(steps, AddPeer{ToStore: id, PeerID: peer.Id})
+		}
+	}
+
+	// Remove redundant peers.
+	for _, peer := range region.GetPeers() {
+		if _, ok := storeIDs[peer.GetStoreId()]; ok {
+			continue
+		}
+		steps = append(steps, RemovePeer{FromStore: peer.GetStoreId()})
```
Comment (nolouch) [inline server/schedule/operator.go]: `TiKV` will reject remove the leader peer now.

File: server/schedule/operator.go (hunk 1 above)
Comment (rleungx) [reply server/schedule/operator.go]: To keep consistent with others, I think it's ok.

File: server/schedule/operator.go [hunk 4]
```diff
@@ -473,6 +540,27 @@ func CreateMovePeerOperator(desc string, cluster Cluster, region *core.RegionInf
...
 
+// CreateMoveLeaderOperator creates an operator that replaces an old leader with a new leader.
+func CreateMoveLeaderOperator(desc string, cluster Cluster, region *core.RegionInfo, kind OperatorKind, oldStore, newStore uint64, peerID uint64) (*Operator, error) {
+	removeKind, steps, err := removePeerSteps(cluster, region, oldStore, []uint64{newStore})
+	if err != nil {
+		return nil, err
+	}
+	st := CreateAddPeerSteps(newStore, peerID, cluster)
+	steps = append(st, steps...)
+	return NewOperator(desc, region.GetID(), region.GetRegionEpoch(), removeKind|kind|OpLeader|OpRegion, steps...), nil
```
Comment (disksing) [inline server/schedule/operator.go]: No need to transfer leader to the new peer?

File: server/schedule/operator.go [hunk 5]
```diff
@@ -499,6 +501,27 @@ func (o *Operator) History() []OperatorHistory {
...
+// CreateAddPeerOperator creates an operator that adds a new peer.
+func CreateAddPeerOperator(desc string, cluster Cluster, region *core.RegionInfo, peerID uint64, toStoreID uint64, kind OperatorKind) *Operator {
+	steps := CreateAddPeerSteps(toStoreID, peerID, cluster)
+	return NewOperator(desc, region.GetID(), region.GetRegionEpoch(), kind|OpRegion, steps...)
+}
+
+// CreateAddLearnerOperator creates an operator that adds a new learner.
+func CreateAddLearnerOperator(desc string, cluster Cluster, region *core.RegionInfo, peerID uint64, toStoreID uint64, kind OperatorKind) *Operator {
+	steps := []OperatorStep{AddLearner{ToStore: toStoreID, PeerID: peerID}}
+	return NewOperator(desc, region.GetID(), region.GetRegionEpoch(), kind|OpRegion, steps...)
```
Comment (shafreeck) [inline server/schedule/operator.go]: There is only one step here, should this array be eliminated.

File: server/schedule/operator.go [hunk 6]
```diff
@@ -540,6 +563,50 @@ func CreateAddLightPeerSteps(newStore uint64, peerID uint64, cluster Cluster) []
 	return st
 }
 
+// CreateTransferLeaderOperator creates an operator that transfers the leader from a source store to a target store.
+func CreateTransferLeaderOperator(desc string, region *core.RegionInfo, sourceStoreID uint64, targetStoreID uint64, kind OperatorKind) *Operator {
+	steps := TransferLeader{FromStore: sourceStoreID, ToStore: targetStoreID}
```
Comment (shafreeck) [inline server/schedule/operator.go]:
```suggestion
	step := TransferLeader{FromStore: sourceStoreID, ToStore: targetStoreID}
```

**Pull request merged**